docs/30DaysOfPWA/advanced-capabilities/06.md

```diff
@@ -1,15 +1,174 @@
+
+- Capping the number of background sync retries and duration of periodic syncs. It's not defined in API specification, but for the periodic sync intervals, browsers use the _Site Engagement_ index (number and duration of user interactions) for the particular origin on a particular device to calculate the exact intervals. At best it is a few hours. Hint: you can check the Site Engagement index for yourself on your current Chromium-based browser by visiting [about://site-engagement/](about://site-engagement/) url.
+- Code execution time is limited by the service worker's lifetime which is a few seconds. So abusing device hardware by heavy computation or overusing network resources is rather difficult. For downloading the large files and having their content available for your frontend app, you can use [Background Fetch API](https://docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/background-syncs#use-the-background-fetch-api-to-fetch-large-files-when-the-app-or-service-worker-isnt-running) from the same background services family. This API is designed to keep the user in full control of the long network operation by displaying a browser native UI component with the download progress and Pause/Cancel/Resume buttons.
+- Both `sync` and `periodicsync` events are only fired online. For Background Sync detecting online is a main and only purpose. While for Periodic Background Sync this limitation emphasizes the intended use case: keeping the application itself and/or its data up-to-date by periodically syncing with the server.
+
+Periodic Background Sync API has a few additional requirements:
+
+- The `periodicsync` will happen only on a known network (the one user was connected to before).
+- It only works for _installed_ web applications.
+
+Please note, that the limitations listed above are a current "snapshot" which is a mix of MAY, SHOULD, MUST statements of the respective specifications and some additional voluntary steps from the browser vendors.
+
+## Testing and debugging
+
+Checking how the background services you implemented work might be not an easy task: you as a developer do not have full control over the exact times when sync events come to a service worker. Fortunately, we have a set of helpers in the Chromium-based browser DevTools.
+
+Go to Application -> Service Workers pane to send `sync` and `periodicsync` events to the service worker manually:
+
+![Sending events manually](_media/background-sync-devtools-send.png)
+
+In the Background Services section of the Application tab, you can click on the API you want to debug to see the events (both "real" and "manual") that were sent to the service worker. To "catch" and debug the real ones (distributed over hours and days if we talk about periodic syncs), you can use the "Record" icon to log the events for up to three days.
+
+![Logging background service events](_media/background-sync-devtools-record.png)
```

Comment:
Week 3 days 3 and 4 already go into a lot of debugging details. Day 4 in particular is about debugging advanced capabilities. So we might want to link to these days instead of embedding the content here. This way this article can be shorter and more focused on code examples.

Reply:
Fully agree. The only challenge is - it's in Week 3 so this chapter will not be published yet. Maybe, saying "you will learn how to debug next week" could sound strange then :)

We can remove this short part and change it to link to Week 3 Day 3-4 after they published

Reply:
Ah good idea! I like it. Let's keep it in and then modify to link to week 3 content when it's available.